File: web/src/render/listTaskItemRenderer.ts
import type MarkdownNode from "../types/MarkdownNode";
import type Renderer from "../types/Renderer";
import type RendererState from "../types/RendererState";
import isSpace from "../utils/isSpace";

const renderer: Renderer = {
	name: "list_task_item",
	render,
};
export default renderer;

function render(node: MarkdownNode, state: RendererState): void {
	let checked = !isSpace(node.markup.charCodeAt(1)) ? ` checked=""` : ``;
	state.output += `<input type="checkbox"${checked} disabled="" /> `;
}

File: zig/src/render/listTaskItemRenderer.zig
const std = @import("std");
const isSpace = @import("../utils/isSpace.zig").isSpace;

const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;

pub fn render(node: *const MarkdownNode, state: *RendererState, first: ?bool, last: ?bool, decode: ?bool) void {
    _ = first;
    _ = last;
    _ = decode;

    const checked = if (node.markup.len > 1 and !isSpace(node.markup[1]))
        " checked=\"\""
    else
        "";

    state.output.writer().print("<input type=\"checkbox\"{s} disabled=\"\" /> ", .{checked}) catch unreachable;
}

pub const listTaskItemRenderer = Renderer{
    .name = "list_task_item",
    .render = render,
};
